"use strict";(self.webpackChunkmatic_docs=self.webpackChunkmatic_docs||[]).push([[75643],{3905:function(e,t,n){n.d(t,{Zo:function(){return d},kt:function(){return m}});var r=n(67294);function a(e,t,n){return t in e?Object.defineProperty(e,t,{value:n,enumerable:!0,configurable:!0,writable:!0}):e[t]=n,e}function o(e,t){var n=Object.keys(e);if(Object.getOwnPropertySymbols){var r=Object.getOwnPropertySymbols(e);t&&(r=r.filter((function(t){return Object.getOwnPropertyDescriptor(e,t).enumerable}))),n.push.apply(n,r)}return n}function i(e){for(var t=1;t<arguments.length;t++){var n=null!=arguments[t]?arguments[t]:{};t%2?o(Object(n),!0).forEach((function(t){a(e,t,n[t])})):Object.getOwnPropertyDescriptors?Object.defineProperties(e,Object.getOwnPropertyDescriptors(n)):o(Object(n)).forEach((function(t){Object.defineProperty(e,t,Object.getOwnPropertyDescriptor(n,t))}))}return e}function l(e,t){if(null==e)return{};var n,r,a=function(e,t){if(null==e)return{};var n,r,a={},o=Object.keys(e);for(r=0;r<o.length;r++)n=o[r],t.indexOf(n)>=0||(a[n]=e[n]);return a}(e,t);if(Object.getOwnPropertySymbols){var o=Object.getOwnPropertySymbols(e);for(r=0;r<o.length;r++)n=o[r],t.indexOf(n)>=0||Object.prototype.propertyIsEnumerable.call(e,n)&&(a[n]=e[n])}return a}var s=r.createContext({}),p=function(e){var t=r.useContext(s),n=t;return e&&(n="function"==typeof e?e(t):i(i({},t),e)),n},d=function(e){var t=p(e.components);return r.createElement(s.Provider,{value:t},e.children)},u={inlineCode:"code",wrapper:function(e){var t=e.children;return r.createElement(r.Fragment,{},t)}},c=r.forwardRef((function(e,t){var n=e.components,a=e.mdxType,o=e.originalType,s=e.parentName,d=l(e,["components","mdxType","originalType","parentName"]),c=p(n),m=a,k=c["".concat(s,".").concat(m)]||c[m]||u[m]||o;return n?r.createElement(k,i(i({ref:t},d),{},{components:n})):r.createElement(k,i({ref:t},d))}));function m(e,t){var n=arguments,a=t&&t.mdxType;if("string"==typeof e||a){var o=n.length,i=new Array(o);i[0]=c;var l={};for(var s in t)hasOwnProperty.call(t,s)&&(l[s]=t[s]);l.originalType=e,l.mdxType="string"==typeof e?e:a,i[1]=l;for(var p=2;p<o;p++)i[p]=n[p];return r.createElement.apply(null,i)}return r.createElement.apply(null,n)}c.displayName="MDXCreateElement"},79779:function(e,t,n){n.r(t),n.d(t,{assets:function(){return u},contentTitle:function(){return p},default:function(){return k},frontMatter:function(){return s},metadata:function(){return d},toc:function(){return c}});var r=n(87462),a=n(63366),o=(n(67294),n(3905)),i=n(44996),l=["components"],s={id:"known-issues",title:"Known Issues & Errors",description:"Knowledge Base of common errors",keywords:["docs","polygon","matic","issues","errors","invalid","command"],slug:"known-issues",image:"https://wiki.polygon.technology/img/polygon-logo.png"},p=void 0,d={unversionedId:"maintain/validate/kb/known-issues",id:"maintain/validate/kb/known-issues",title:"Known Issues & Errors",description:"Knowledge Base of common errors",source:"@site/docs/maintain/validate/kb/known-issues.md",sourceDirName:"maintain/validate/kb",slug:"/maintain/validate/kb/known-issues",permalink:"/matic-docs/docs/maintain/validate/kb/known-issues",draft:!1,editUrl:"https://github.com/maticnetwork/matic-docs/tree/master/docs/maintain/validate/kb/known-issues.md",tags:[],version:"current",lastUpdatedBy:"k-batista",lastUpdatedAt:1686626387,formattedLastUpdatedAt:"Jun 13, 2023",frontMatter:{id:"known-issues",title:"Known Issues & Errors",description:"Knowledge Base of common errors",keywords:["docs","polygon","matic","issues","errors","invalid","command"],slug:"known-issues",image:"https://wiki.polygon.technology/img/polygon-logo.png"},sidebar:"maintain",previous:{title:"Knowledge Base",permalink:"/matic-docs/docs/category/knowledge-base"},next:{title:"Pre-requisites & How Tos",permalink:"/matic-docs/docs/maintain/validate/kb/how-to"}},u={},c=[{value:"Error: Bad block/Invalid Merkle",id:"error-bad-blockinvalid-merkle",level:3},{value:"Log: Error validating checkpoint module=checkpoint startBlock",id:"log-error-validating-checkpoint-modulecheckpoint-startblock",level:3},{value:"Log: Error dialing seed",id:"log-error-dialing-seed",level:3},{value:"Log: Demoting invalidated transaction",id:"log-demoting-invalidated-transaction",level:3},{value:"Error: Failed Sanity Checks",id:"error-failed-sanity-checks",level:3},{value:"Issue: Bor synchronisation is slow",id:"issue-bor-synchronisation-is-slow",level:3},{value:"Node is not signing any checkpoints",id:"node-is-not-signing-any-checkpoints",level:3},{value:"Issue: Validator Heimdall is unable to connect to Peers",id:"issue-validator-heimdall-is-unable-to-connect-to-peers",level:3},{value:"Error: Error while fetching mainchain receipt error",id:"error-error-while-fetching-mainchain-receipt-error",level:3},{value:"Validator bor is stuck on block for a long time",id:"validator-bor-is-stuck-on-block-for-a-long-time",level:3},{value:"Error: (in Bor) &quot;Failed to prepare header mining at block 0&quot;",id:"error-in-bor-failed-to-prepare-header-mining-at-block-0",level:3},{value:"Error: &quot;30303&quot; or invalid command",id:"error-30303-or-invalid-command",level:3},{value:"Error: Impossible reorg, please file an issue",id:"error-impossible-reorg-please-file-an-issue",level:3},{value:"Error: &quot;Host not found&quot; while setting up a node using Ansible",id:"error-host-not-found-while-setting-up-a-node-using-ansible",level:3},{value:"Issue: &quot;Dialling failed&quot; in Heimdall",id:"issue-dialling-failed-in-heimdall",level:3},{value:"Issue: Looking for Peers or Stopping Peer for error",id:"issue-looking-for-peers-or-stopping-peer-for-error",level:3},{value:"Error: Error while fetching data from URL",id:"error-error-while-fetching-data-from-url",level:3},{value:"Error: no contract code at the given address",id:"error-no-contract-code-at-the-given-address",level:3},{value:"Issue: Problems in starting Bor",id:"issue-problems-in-starting-bor",level:3},{value:"Error: Failed to unlock account (0x...) No key for given address or file",id:"error-failed-to-unlock-account-0x-no-key-for-given-address-or-file",level:3},{value:"Consequences of validator missing a checkpoint and points to investigate from our side",id:"consequences-of-validator-missing-a-checkpoint-and-points-to-investigate-from-our-side",level:3},{value:"Error: dpkg: error processing archive matic-heimdall-xxxxxxxxxx",id:"error-dpkg-error-processing-archive-matic-heimdall-xxxxxxxxxx",level:3},{value:"Issue: Tendermint was rest without resetting application&#39;s data",id:"issue-tendermint-was-rest-without-resetting-applications-data",level:3},{value:"Issue: Bor crashed",id:"issue-bor-crashed",level:3},{value:"Error: err=&quot;insufficient funds for gas * price + value&quot;",id:"error-errinsufficient-funds-for-gas--price--value",level:3},{value:"Heimdall:  No staking sequence exists: %s %s             module=staking",id:"heimdall--no-staking-sequence-exists-s-s-------------modulestaking",level:3},{value:"Retrying again in 5 seconds to fetch data from Heimdall path=bor/span/1",id:"retrying-again-in-5-seconds-to-fetch-data-from-heimdall-pathborspan1",level:3},{value:"etherbase missing: etherbase must be explicitly specified",id:"etherbase-missing-etherbase-must-be-explicitly-specified",level:3},{value:"Steps to Prune the node",id:"steps-to-prune-the-node",level:3}],m={toc:c};function k(e){var t=e.components,n=(0,a.Z)(e,l);return(0,o.kt)("wrapper",(0,r.Z)({},m,n,{components:t,mdxType:"MDXLayout"}),(0,o.kt)("admonition",{type:"info"},(0,o.kt)("p",{parentName:"admonition"},"If you have any queries about the Validator process or facing issues with Heimdall and Bor, head over to the ",(0,o.kt)("a",{parentName:"p",href:"/matic-docs/docs/faq/validator-faq"},(0,o.kt)("ins",null,"Validator FAQ"))," page.")),(0,o.kt)("h3",{id:"error-bad-blockinvalid-merkle"},"Error: Bad block/Invalid Merkle"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:"),"\nA Bad Block or Invalid Merkle root error occurs when your Heimdall and Bor are not in sync with each other. Heimdall is the consensus layer for Polygon POS chain, which means that Heimdall directs Bor to create blocks accordingly. A Bad Block occurs when Bor moves ahead to create a block which has not been directed by Heimdall and hence there is an invalid hash been created, which causes the error, Bad Block, or Invalid Merkle root."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution 1"),":\nTypically a restart of the Bor service should resolve the problem. This would ensure your Bor connects with Heimdall again and start syncing and creating blocks correctly."),(0,o.kt)("p",null,"To restart your Bor service you can use the following command, ",(0,o.kt)("inlineCode",{parentName:"p"},"sudo service bor restart"),"."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution 2"),":\nIf a restart does not fix your problem, then the first thing you need to check is your Heimdall and REST server. Most of the time, your Heimdall service might have stopped which has caused the Bad block issue on Bor."),(0,o.kt)("p",null,"Check the logs for your Heimdall first, ",(0,o.kt)("inlineCode",{parentName:"p"},"journalctl -u heimdalld -f")," and check if everything is working correctly."),(0,o.kt)("p",null,"Additionally, also check your REST server logs, ",(0,o.kt)("inlineCode",{parentName:"p"},"journalctl -u heimdalld-rest-server -f"),"."),(0,o.kt)("p",null,"If you find that any of these services are not running correctly, then please restart the above services and let them sync. Your Bor should automatically resolve the problem."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution 3"),":\nIf a restart of your Bor and Heimdall services don't resolve the problem, then its probably that your Bor is stuck on a block. The block number will be evident in the logs. To check your logs for Bor you can run this command, ",(0,o.kt)("inlineCode",{parentName:"p"},"journalctl -u bor -f"),"."),(0,o.kt)("p",null,"The Bad block would be displayed this way in your logs:"),(0,o.kt)("img",{src:(0,i.Z)("img/knowledge-base/bad_block.png")}),(0,o.kt)("p",null,"Once you know the Bad block number, you could roll back your Blockchain by a few hundred blocks and resync from a previous block. In order to do this, you will first need to convert the Block number to hexadecimal. You can use ",(0,o.kt)("a",{parentName:"p",href:"https://www.rapidtables.com/convert/number/decimal-to-hex.html"},"this tool")," for converting decimals to hexadecimals."),(0,o.kt)("p",null,"Once you have your Hexadecimal ready, you can run the following commands;"),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-bash"},'bor attach ./.bor/data/bor.ipc\n> debug.setHead("0xE92570")\n')),(0,o.kt)("p",null,(0,o.kt)("inlineCode",{parentName:"p"},"debug.setHead()")," is the function that will allow your Bor to set the tip at a particular Block height."),(0,o.kt)("p",null,"Once you run these commands, the output for this would be ",(0,o.kt)("inlineCode",{parentName:"p"},"null")," . Null means good and it is intended. You can now start monitoring your logs for Bor again and see if it passes that block number."),(0,o.kt)("p",null,"If in any case, none of these solutions work for you, please contact the Polygon Support team immediately."),(0,o.kt)("h3",{id:"log-error-validating-checkpoint-modulecheckpoint-startblock"},"Log: Error validating checkpoint module=checkpoint startBlock"),(0,o.kt)("p",null,"If your node throws up these logs, please check the following:"),(0,o.kt)("ul",null,(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"Ensure that the ",(0,o.kt)("strong",{parentName:"p"},"Bor node is in sync"),". To check, run the following command:"),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre",className:"language-bash"},"bor attach .bor/data/bor.ipc\neth.syncing\n")),(0,o.kt)("p",{parentName:"li"},"  If the output is ",(0,o.kt)("strong",{parentName:"p"},"false"),", then the Bor node is in sync.")),(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"Another possible cause of this issue is that ",(0,o.kt)("strong",{parentName:"p"},"your node might be on the wrong fork"),". To check, run the following command to find the current block on your node:"),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre",className:"language-bash"},"bor attach .bor/data/bor.ipc\neth_blockNumber\n")),(0,o.kt)("p",{parentName:"li"},"  Find the associated block hash for the block number:"),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre"},'bor attach .bor/data/bor.ipc\neth.getBlockByNumber("<Block Number>").hash\n')),(0,o.kt)("p",{parentName:"li"},"  Now, you can inspect the block number to identify if you are running on the right fork. One way to do this is to search for the block number on an explorer like ",(0,o.kt)("a",{parentName:"p",href:"https://polygonscan.com/"},"PolygonScan"),"."),(0,o.kt)("img",{src:(0,i.Z)("img/knowledge-base/block_number.png")}),(0,o.kt)("p",{parentName:"li"},"  If the hashes match, then the node is on the right fork."))),(0,o.kt)("h3",{id:"log-error-dialing-seed"},"Log: Error dialing seed"),(0,o.kt)("p",null,"Please check whether your Heimdall node is configured with the latest seeds as listed on the ",(0,o.kt)("a",{parentName:"p",href:"/matic-docs/docs/operate/full-node-deployment#full-node-setup"},"node setup documents"),"."),(0,o.kt)("p",null,"If you're still encountering the error after either updating to the latest seeds or confirming that you are using the right seeds, you may need to clear the ",(0,o.kt)("inlineCode",{parentName:"p"},"addrbook.json")," file. To do this, follow the steps below."),(0,o.kt)("ol",null,(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"Open the ",(0,o.kt)("inlineCode",{parentName:"p"},"config.toml")," file in your terminal: ",(0,o.kt)("inlineCode",{parentName:"p"},"vi /var/lib/heimdall/config/config.toml"))),(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"Stop ",(0,o.kt)("inlineCode",{parentName:"p"},"heimdalld")," service: ",(0,o.kt)("inlineCode",{parentName:"p"},"sudo service heimdalld stop"))),(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"Clear your ",(0,o.kt)("inlineCode",{parentName:"p"},"addrbook")),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre"},"sudo service heimdalld stop\ncp /var/lib/heimdall/config/addrbook.json /var/lib/heimdall/config/addrbook.json.bkp\nrm /var/lib/heimdall/config/addrbook.json\n"))),(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"Increase ",(0,o.kt)("inlineCode",{parentName:"p"},"max_num_inbound_peers")," and ",(0,o.kt)("inlineCode",{parentName:"p"},"max_num_outbound_peers")," in ",(0,o.kt)("inlineCode",{parentName:"p"},"/var/lib/heimdall/config/config.toml"),":"),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre"},"max_num_inbound_peers = 300\nmax_num_outbound_peers = 100\n"))),(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"Start ",(0,o.kt)("inlineCode",{parentName:"p"},"heimdalld")," service: ",(0,o.kt)("inlineCode",{parentName:"p"},"sudo service heimdalld start")))),(0,o.kt)("h3",{id:"log-demoting-invalidated-transaction"},"Log: Demoting invalidated transaction"),(0,o.kt)("p",null,"This log is not an error. It is a process in ",(0,o.kt)("strong",{parentName:"p"},"txpool")," which rearranges the transactions and removes some of them (as per specific conditions)."),(0,o.kt)("p",null,"It should in ",(0,o.kt)("strong",{parentName:"p"},"no way affect the checkpointing mechanism"),"."),(0,o.kt)("h3",{id:"error-failed-sanity-checks"},"Error: Failed Sanity Checks"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:"),"\n",(0,o.kt)("inlineCode",{parentName:"p"},"Addressbook")," warnings can be ignored without an issue most of the time. If your node is connected to sufficient number of peers these kind of errors can be ignored. Your ",(0,o.kt)("inlineCode",{parentName:"p"},"pex")," is just trying to re-establish it's connections with peers already present in ",(0,o.kt)("inlineCode",{parentName:"p"},"addrbook.json"),"."),(0,o.kt)("h3",{id:"issue-bor-synchronisation-is-slow"},"Issue: Bor synchronisation is slow"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:"),"\nIf Bor synchronisation is slow it could be due to either of the below reasons:"),(0,o.kt)("ul",null,(0,o.kt)("li",{parentName:"ul"},"The node is running on a fork - means at certain point the block production was done by forking on a different block and that has impacted the further block production"),(0,o.kt)("li",{parentName:"ul"},"The machine is not working at optimum levels and could be with insufficient resources.",(0,o.kt)("ul",{parentName:"li"},(0,o.kt)("li",{parentName:"ul"},"This can be addressed by checking on:",(0,o.kt)("ul",{parentName:"li"},(0,o.kt)("li",{parentName:"ul"},"IOPS",(0,o.kt)("ul",{parentName:"li"},(0,o.kt)("li",{parentName:"ul"},"IOPS stands for Input/Output state of cycle"),(0,o.kt)("li",{parentName:"ul"},"The rate of reading is usually higher than write speed"),(0,o.kt)("li",{parentName:"ul"},"6000 is the recommended range for IOPS"))),(0,o.kt)("li",{parentName:"ul"},"Processing Power",(0,o.kt)("ul",{parentName:"li"},(0,o.kt)("li",{parentName:"ul"},"Processor has to be 8 or 16 core"),(0,o.kt)("li",{parentName:"ul"},"RAM: 32 GB is the minimum; 64 GB is recommended"),(0,o.kt)("li",{parentName:"ul"},"Block import should be more than 2 block for every second"))),(0,o.kt)("li",{parentName:"ul"},"Node sync rate should be at 15-20 blocks every 8 secs")))))),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:"),"\nAs the issue is more about lack of hardware resources try upgrading it to double of the current specifications."),(0,o.kt)("h3",{id:"node-is-not-signing-any-checkpoints"},"Node is not signing any checkpoints"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"PreRequisite:"),"\nFirst, please point the bor_rpc_url(heimdall-config.toml) of the validator to any external infra providers and restart the services. This change helps to avoid missing checkpoints."),(0,o.kt)("p",null,"You can find a list of Infra Providers, on the navbar, that validators can make use of."),(0,o.kt)("admonition",{type:"note"},(0,o.kt)("p",{parentName:"admonition"},"At this point in time, the node will not mine blocks. So once the issue is fixed, the changes made have to be reverted for the node to return to normal functionality.")),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:"),"\nFirst of all, your node not signing checkpoints could be for a multiple reasons."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution 1:"),"\nFirst check if your Heimdall service is running correctly on your Sentry and Validator node. If the service has stopped abruptly or see any errors, try restarting your Heimdall service and see it comes back to normal."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution 2:"),"\nCheck your Bor service and see if it has halted abruptly or there are any errors on the logs. Try restarting your Bor service to resolve this issue."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution 3:"),"\nCheck if your Heimdall Bridge is running or not or if it has any errors in the logs. Try restarting the service and see if the issue resolves."),(0,o.kt)("h3",{id:"issue-validator-heimdall-is-unable-to-connect-to-peers"},"Issue: Validator Heimdall is unable to connect to Peers"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:"),"\nThis typically means that your Sentry Heimdall is running into issues."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:")),(0,o.kt)("ul",null,(0,o.kt)("li",{parentName:"ul"},"Check your Sentry Heimdall and see if the service is running fine."),(0,o.kt)("li",{parentName:"ul"},"If the service is stopped then restarting the service on your Sentry should resolve this issue."),(0,o.kt)("li",{parentName:"ul"},"Similarly, after fixing your sentry, a restart of your Heimdall service should also resolve the problem.")),(0,o.kt)("h3",{id:"error-error-while-fetching-mainchain-receipt-error"},"Error: Error while fetching mainchain receipt error"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:")," These are normal logs. Do not do anything to your bridge."),(0,o.kt)("h3",{id:"validator-bor-is-stuck-on-block-for-a-long-time"},"Validator bor is stuck on block for a long time"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:"),"\nThis means that your Bor on your Sentry is also stuck because your Validator gets information from your Sentry."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:")),(0,o.kt)("ul",null,(0,o.kt)("li",{parentName:"ul"},"Please check your Bor logs on your sentry and see if everything is okay."),(0,o.kt)("li",{parentName:"ul"},"Probably restart the Bor service first on your Bor and then simultaneously restart the Bor service on your Validator as well.")),(0,o.kt)("h3",{id:"error-in-bor-failed-to-prepare-header-mining-at-block-0"},'Error: (in Bor) "Failed to prepare header mining at block 0"'),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:"),"\nThis happens because of a formatting issue in your ",(0,o.kt)("inlineCode",{parentName:"p"},"static-nodes.json")," file (/var/lib/bor/data/bor/static-nodes.json)."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:")),(0,o.kt)("ul",null,(0,o.kt)("li",{parentName:"ul"},"Ensure there are no space and no additional characters like < / > ."),(0,o.kt)("li",{parentName:"ul"},"If you have made any changes to the file then please restart your Bor service and you should see logs printing.")),(0,o.kt)("h3",{id:"error-30303-or-invalid-command"},'Error: "30303" or invalid command'),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:"),"\nThis is because you haven\u2019t created the bor keystore and the password file for it."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:")),(0,o.kt)("p",null,"Ensure that you follow all the steps from the guide setup."),(0,o.kt)("h3",{id:"error-impossible-reorg-please-file-an-issue"},"Error: Impossible reorg, please file an issue"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:"),"\nLet these logs be. Your node should ideally not suffer because of this and the issue should be automatically resolved."),(0,o.kt)("p",null,"If your node is suffering because of this, please contact the support team immdiately."),(0,o.kt)("h3",{id:"error-host-not-found-while-setting-up-a-node-using-ansible"},'Error: "Host not found" while setting up a node using Ansible'),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:"),"\nThis could be because your ",(0,o.kt)("inlineCode",{parentName:"p"},"inventory.yml")," file may have some formatting issues."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:"),"\nCorrect them with proper indentation and then try again."),(0,o.kt)("h3",{id:"issue-dialling-failed-in-heimdall"},'Issue: "Dialling failed" in Heimdall'),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:"),"\nThis is related to connectivity and more specifically a port related problem."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:")),(0,o.kt)("ul",null,(0,o.kt)("li",{parentName:"ul"},"Check to ",(0,o.kt)("inlineCode",{parentName:"li"},"curl localhost:26657/status")," still shows the same block."),(0,o.kt)("li",{parentName:"ul"},"Try a Heimdall Restart."),(0,o.kt)("li",{parentName:"ul"},"Make sure that the connectivity to this port 26656 is open."),(0,o.kt)("li",{parentName:"ul"},"Try adding additional peers in vi /var/lib/heimdall/config/config.toml"),(0,o.kt)("li",{parentName:"ul"},"Set ",(0,o.kt)("inlineCode",{parentName:"li"},"max_open_connection")," parameter to 100.")),(0,o.kt)("h3",{id:"issue-looking-for-peers-or-stopping-peer-for-error"},"Issue: Looking for Peers or Stopping Peer for error"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:")),(0,o.kt)("ul",null,(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"open the\xa0",(0,o.kt)("inlineCode",{parentName:"p"},"config.toml"),"\xa0file on your Sentry node."),(0,o.kt)("p",{parentName:"li"},"  ",(0,o.kt)("inlineCode",{parentName:"p"},"/var/lib/heimdall/config/config.toml"))),(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"And then find the parameter\xa0",(0,o.kt)("inlineCode",{parentName:"p"},"external_address"),". Once you find it this what you should be updating it with"),(0,o.kt)("p",{parentName:"li"},"  ",(0,o.kt)("inlineCode",{parentName:"p"},"tcp://<my_elastic_ip>:26656"))),(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"Where\xa0",(0,o.kt)("inlineCode",{parentName:"p"},"my_elastic_ip"),"\xa0is your Sentry\u2019s public IP")),(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"Once you have updated this, all you need to do is restart your Heimdall service on your Sentry"),(0,o.kt)("p",{parentName:"li"},"  ",(0,o.kt)("inlineCode",{parentName:"p"},"sudo service heimdalld restart"))),(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"Ensure that you\u2019re only doing this on your sentry only."))),(0,o.kt)("p",null,"Follow the below steps for adding additional peers in  ",(0,o.kt)("inlineCode",{parentName:"p"},"vi /var/lib/heimdall/config/config.toml")),(0,o.kt)("ul",null,(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"Stop heimdalld service"),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre"},"sudo service heimdalld stop\n"))),(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"Clear your ",(0,o.kt)("inlineCode",{parentName:"p"},"addrbook")),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre"},"sudo service heimdalld stop\ncp /var/lib/heimdall/config/addrbook.json /var/lib/heimdall/config/addrbook.json.bkp\nrm /var/lib/heimdall/config/addrbook.json\n"))),(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"Increase ",(0,o.kt)("inlineCode",{parentName:"p"},"max_num_inbound_peers")," and ",(0,o.kt)("inlineCode",{parentName:"p"},"max_num_outbound_peers")," in\xa0",(0,o.kt)("inlineCode",{parentName:"p"},"/var/lib/heimdall/config/config.toml"),":"),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre"},"max_num_inbound_peers = 300\nmax_num_outbound_peers = 100\n"))),(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"Start heimdalld service:"),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre"},"sudo service heimdalld start\n")))),(0,o.kt)("h3",{id:"error-error-while-fetching-data-from-url"},"Error: Error while fetching data from URL"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Error sample:")),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-bash"},'module=span service=processor error="Error while fetching data from url:\n[http://0.0.0.0:1317/bor/prepare-next-span?chain_id=137&proposer=0x29f265b54a298df0c1b762f688e7e7c09d8790ea&span_id=2863&start_block=18317056](http://0.0.0.0:1317/bor/prepare-next-span?chain_id=137&proposer=0x29f265b54a298df0c1b762f688e7e7c09d8790ea&span_id=2863&start_block=18317056), status: 400"\nAug 23 12:07:23 US-CA-SN01 bridge[2340]: E[2021-08-23|12:07:23.158] Unable to fetch next span details\nmodule=span service=processor lastSpanId=2862\n')),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:")),(0,o.kt)("p",null,"Then the Heimdall Bridge needs a restart."),(0,o.kt)("h3",{id:"error-no-contract-code-at-the-given-address"},"Error: no contract code at the given address"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution")),(0,o.kt)("ol",null,(0,o.kt)("li",{parentName:"ol"},"Get the right configs from Github and copy them to ",(0,o.kt)("inlineCode",{parentName:"li"},"/var/lib/heimdall/config")," and"),(0,o.kt)("li",{parentName:"ol"},"Please reset heimdall using ",(0,o.kt)("inlineCode",{parentName:"li"},"heimdalld unsafe-reset-all"),".")),(0,o.kt)("h3",{id:"issue-problems-in-starting-bor"},"Issue: Problems in starting Bor"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Issue:"),"\nAddress is required as an argument."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:"),"\nYou have to add address."),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-bash"},"/etc/matic/metadata\n")),(0,o.kt)("h3",{id:"error-failed-to-unlock-account-0x-no-key-for-given-address-or-file"},"Error: Failed to unlock account (0x...) No key for given address or file"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:")),(0,o.kt)("p",null,"This error happens in light of the fact that the way for the password.txt record is erroneous. You can follow the beneath steps to amend this."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:")),(0,o.kt)("p",null,"For Linux packages:"),(0,o.kt)("p",null,"Kill Bor process"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"For linux"),":"),(0,o.kt)("ol",null,(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("inlineCode",{parentName:"li"},"ps -aux | grep bor"),". Get the PID for Bor and then run the following command."),(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("inlineCode",{parentName:"li"},"sudo kill -9 PID"))),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"For Ansible:")),(0,o.kt)("ol",null,(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"Copy the bor keystore file to"),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre",className:"language-jsx"},"/etc/bor/dataDir/keystore\n"))),(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"And password.txt to"),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre",className:"language-jsx"},"/etc/bor/dataDir/\n"))),(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"Make sure you have added correct address in\xa0",(0,o.kt)("inlineCode",{parentName:"p"},"/etc/bor/metadata")))),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"For Binaries:")),(0,o.kt)("ol",null,(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"Copy the Bor keystore file to:"),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre",className:"language-jsx"},"/var/lib/bor/keystore/\n"))),(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"And password.txt to"),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre",className:"language-jsx"},"/var/lib/bor/password.txt\n")))),(0,o.kt)("h3",{id:"consequences-of-validator-missing-a-checkpoint-and-points-to-investigate-from-our-side"},"Consequences of validator missing a checkpoint and points to investigate from our side"),(0,o.kt)("ul",null,(0,o.kt)("li",{parentName:"ul"},"Economics",(0,o.kt)("ul",{parentName:"li"},(0,o.kt)("li",{parentName:"ul"},"Bad reputation for Validator"),(0,o.kt)("li",{parentName:"ul"},"Rewards gets missed for Delegator"))),(0,o.kt)("li",{parentName:"ul"},"Investigation",(0,o.kt)("ul",{parentName:"li"},(0,o.kt)("li",{parentName:"ul"},"Ask for recent logs")))),(0,o.kt)("h3",{id:"error-dpkg-error-processing-archive-matic-heimdall-xxxxxxxxxx"},"Error: dpkg: error processing archive matic-heimdall-xxxxxxxxxx"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Sample:")),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-bash"}," \"dpkg: error processing archive matic-heimdall_1.0.0_amd64.deb (--install): trying to overwrite '/heimdalld-rest-server.service', which is also in package matic-node 1.0.0\"\n")),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:")),(0,o.kt)("p",null,"This occurs mainly because of a previous installation of Matic on machine. To resolve you can run: ",(0,o.kt)("inlineCode",{parentName:"p"},"sudo dpkg -r matic-node")),(0,o.kt)("h3",{id:"issue-tendermint-was-rest-without-resetting-applications-data"},"Issue: Tendermint was rest without resetting application's data"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:")),(0,o.kt)("ul",null,(0,o.kt)("li",{parentName:"ul"},(0,o.kt)("p",{parentName:"li"},"Reset Heimdall config data and try running the installation again;"),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre",className:"language-jsx"},"$ heimdalld unsafe-reset-all\n")),(0,o.kt)("pre",{parentName:"li"},(0,o.kt)("code",{parentName:"pre",className:"language-jsx"},"$ rm -rf $HEIMDALLDIR/bridge\n")))),(0,o.kt)("h3",{id:"issue-bor-crashed"},"Issue: Bor crashed"),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:")),(0,o.kt)("ul",null,(0,o.kt)("li",{parentName:"ul"},"Try upgrading to double the amount of RAM"),(0,o.kt)("li",{parentName:"ul"},"For example, their current RAM capacity is 16GB, it can be upgraded to 32GB")),(0,o.kt)("h3",{id:"error-errinsufficient-funds-for-gas--price--value"},'Error: err="insufficient funds for gas * price + value"'),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Description:")),(0,o.kt)("p",null,"These logs throw up when there is no enough ETH in your signer wallet."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Solution:"),"\nIt is recommended to have 1 ETH in your signer wallet but can keep .5 to .75 in case you check it often enough."),(0,o.kt)("h3",{id:"heimdall--no-staking-sequence-exists-s-s-------------modulestaking"},"Heimdall:  No staking sequence exists: %s %s             module=staking"),(0,o.kt)("p",null,"If the following logs are found on a large frequency on Heimdall then this issue is related to the bridge service resetting the bridge directory fixes this issue."),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-bash"},"sudo service rabbitmq-server stop\nmv /var/lib/rabbitmq/mnesia /var/lib/rabbitmq/mnesia-old\nsudo service rabbitmq-server start\n")),(0,o.kt)("h3",{id:"retrying-again-in-5-seconds-to-fetch-data-from-heimdall-pathborspan1"},"Retrying again in 5 seconds to fetch data from Heimdall path=bor/span/1"),(0,o.kt)("p",null,"These logs in Bor mean that it cannot connect to Heimdall."),(0,o.kt)("p",null,"The Heimdall doesn\u2019t look in sync and hence it won\u2019t have data on all the things that Bor would require."),(0,o.kt)("p",null,"So the recommended procedure would be to clear the historical data of both Heimdall and Bor and resync from the snapshot."),(0,o.kt)("p",null,"Ensure the following is fine:"),(0,o.kt)("ol",null,(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"Heimdall logs are normal or is it throwing up any errors?")),(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"Ensure Heimdall is fully synced by running: ",(0,o.kt)("inlineCode",{parentName:"p"},"curl localhost:26657/status"))),(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"Also ensure whether Heimdall is connected with the other peers."))),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-bash"},"curl localhost:26657/net_info? | jq .result.n_peers\n")),(0,o.kt)("p",null,"If there aren\u2019t any peers, check whether the ",(0,o.kt)("strong",{parentName:"p"},"seeds or persistent peers are rightly set on Heimdall")," and ",(0,o.kt)("strong",{parentName:"p"},"ensure Port 26656 is all open"),"."),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Reset Heimdall")),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-bash"},"sudo service heimdalld stop\nheimdalld unsafe-reset-all\n")),(0,o.kt)("p",null,(0,o.kt)("strong",{parentName:"p"},"Sync Heimdall from Snapshot")),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-bash"},"wget -c <Snapshot URL>\ntar -xzvf <snapshot file> -C <HEIMDALL_DATA_DIRECTORY>\n")),(0,o.kt)("h3",{id:"etherbase-missing-etherbase-must-be-explicitly-specified"},"etherbase missing: etherbase must be explicitly specified"),(0,o.kt)("p",null,"To fix this issue, the signer address that is used to mine must be added inside ",(0,o.kt)("inlineCode",{parentName:"p"},"miner.etherbase")," section in the ",(0,o.kt)("inlineCode",{parentName:"p"},"config.toml")," file."),(0,o.kt)("h3",{id:"steps-to-prune-the-node"},"Steps to Prune the node"),(0,o.kt)("p",null,"Please use the below steps:"),(0,o.kt)("ol",null,(0,o.kt)("li",{parentName:"ol"},"Check your Bor data size before pruning")),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-bash"},"du -sh /usr/bin/bor\n")),(0,o.kt)("ol",{start:2},(0,o.kt)("li",{parentName:"ol"},"Stop Bor")),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-bash"},"sudo service bor stop\n")),(0,o.kt)("ol",{start:3},(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"Start ",(0,o.kt)("strong",{parentName:"p"},"tmux")," to ensure that even if your SSH connection is reset, the process is running on the remote machine\ntmux.")),(0,o.kt)("li",{parentName:"ol"},(0,o.kt)("p",{parentName:"li"},"Start pruning"))),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-bash"},"sudo bor snapshot prune-state --datadir  /usr/bin/bor\n")),(0,o.kt)("p",null,"The default --datadir is ",(0,o.kt)("inlineCode",{parentName:"p"},"/usr/bin/bor"),"."),(0,o.kt)("ol",{start:5},(0,o.kt)("li",{parentName:"ol"},"Once the pruning is completed, you will see success logs and details. Then start Bor again:")),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-bash"},"sudo service bor start\n")),(0,o.kt)("ol",{start:6},(0,o.kt)("li",{parentName:"ol"},"Check your Bor data size after pruning:")),(0,o.kt)("pre",null,(0,o.kt)("code",{parentName:"pre",className:"language-bash"},"du -sh  /usr/bin/bor\n")))}k.isMDXComponent=!0}}]);